tests/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+include_directories(SYSTEM ${CMAKE_SOURCE_DIR})
+
+enable_testing()
+
+set(INSTALL_GTEST OFF)
+
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)  # cmake-lint: disable=C0103
+
+set(TEST_SOURCES
+        ${CMAKE_SOURCE_DIR}/tests/main.cpp
+        ${CMAKE_SOURCE_DIR}/tests/test_main.cpp
+)
+
+add_executable(sunshine_test ${TEST_SOURCES})
+target_link_libraries(sunshine_test
+        ${SUNSHINE_TARGET_FILES}
+        ${SUNSHINE_EXTERNAL_LIBRARIES}
+        ${EXTRA_LIBS}
+        gtest
+)
+set_target_properties(sunshine_test PROPERTIES CXX_STANDARD 17)
```

Comment:
This won't fix the current linker errors, but your build flags are out of sync with the rest of the project. You should probably add:
`target_compile_options(sunshine_test PRIVATE ${SUNSHINE_COMPILE_OPTIONS})`

Reply:
Thanks, I added that.